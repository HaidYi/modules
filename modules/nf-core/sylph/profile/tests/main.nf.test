nextflow_process {

    name "Test Process SYLPH_PROFILE"
    script "../main.nf"
    process "SYLPH_PROFILE"
    tag "modules"
    tag "modules_nfcore"
    tag "sylph"
    tag "sylph/profile"

    test("sarscov2 illumina single-end [fastq_gz]") {
        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                            [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                input[1] = file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions_single") },
                { assert process.out.profile_out.get(0).size() > 1 }
            )
        }
    }

    test("sarscov2 illumina paired-end [fastq_gz]") {
        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                            [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                input[1] = file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions_paired") },
                { assert process.out.profile_out.get(0).size() > 1 }
            )
        }
    }

    test("sarscov2 illumina paired-end [fastq_gz]-stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [ [ id:'test' ], // meta map
                            [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true)
                    ]
                ]
                input[1] = file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
