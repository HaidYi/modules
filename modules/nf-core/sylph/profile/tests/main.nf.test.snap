{
    "sarscov2 illumina paired-end [fastq_gz]": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.tsv:md5,c1a6fcabeaa1070385f1430d370ba02f"
                ]
            ],
            [
                "versions.yml:md5,7b5a545483277cc0ff9189f8891e737f"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-28T17:29:55.064467"
    },
    "sarscov2 illumina single-end [fastq_gz]": {
        "content": [
            null,
            [
                "versions.yml:md5,7b5a545483277cc0ff9189f8891e737f"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-28T17:27:29.922818"
    },
    "sarscov2 illumina paired-end [fastq_gz]-stub": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "versions.yml:md5,7b5a545483277cc0ff9189f8891e737f"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-28T17:30:02.138682"
    }
}